tests/cache/models/samples/statusCalculation.js

```diff
@@ -0,0 +1,739 @@
+/**
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * tests/cache/models/samples/statusCalculation.js
+ */
+'use strict'; // eslint-disable-line strict
+const chai = require('chai');
+chai.use(require('chai-as-promised')).should();
+const rtu = require('../redisTestUtil');
+const constants = require('../../../../db/constants');
+const redisOps = rtu.redisOps;
+
+describe('tests/cache/models/samples/statusCalculation.js, ' +
+  'redis: aspect: create >', () => {
+
+  afterEach(rtu.flushRedis);
+
+  function setupRanges(ranges) {
+    return redisOps.setRanges({ name: 'asp1', ...ranges });
+  } // setupRanges
+
+  function calculateAndExpect(value, expectedStatus) {
+    return redisOps.calculateSampleStatus('sub1.sub2|asp1', value)
+           .should.eventually.equal(expectedStatus);
+  }
+
+  describe('numeric >', () => {
+    describe('basic ranges (AND calculate invoked on update) >', () => {
+      beforeEach(() =>
+        setupRanges({
+          criticalRange: [0, 1],
+          warningRange: [2, 3],
+          infoRange: [4, 4],
+          okRange: [5, 10],
+        })
+      );
+
+      it('value = range min', () =>
+        calculateAndExpect('2', constants.statuses.Warning)
+      );
+
+      it('value = range max', () =>
+        calculateAndExpect('1', constants.statuses.Critical)
+      );
+
+      it('value = range min and max', () =>
+        calculateAndExpect('4', constants.statuses.Info)
+      );
+
+      it('value is between range min and max', () =>
+        calculateAndExpect('8', constants.statuses.OK)
+      );
+
+      it('value is greater than any of the range max values', () =>
+        calculateAndExpect('11', constants.statuses.Invalid)
+      );
+
+      it('value is less than any of the range min values', () =>
+        calculateAndExpect('-1', constants.statuses.Invalid)
+      );
+    });
+
+    describe('negative int ranges with null range in the middle >', () => {
+      beforeEach(() =>
```

Comment:
this doesn't need to get set up before *each* test in this "describe", just once, right?